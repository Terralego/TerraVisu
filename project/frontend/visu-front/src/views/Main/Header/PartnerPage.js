import React from 'react';
import { Card } from '@blueprintjs/core';


const PartnerPage = () => (
  <Card>
    <div className="mentions-legales">
      <h1>Put your partner data here</h1>
    </div>
  </Card>
);

export default PartnerPage;
