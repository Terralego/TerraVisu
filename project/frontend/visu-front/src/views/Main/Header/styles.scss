@import '../../variables.scss';

.main {

  &__header {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    flex: 0 0 3.35rem;
    z-index: 100;

    &__target {
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      z-index: 105;
    }

    &--mobile {
      flex: 0 0 10rem;
      transform: translateX(0rem);
      margin-left: -7.65rem;
      transition: transform .2s ease-in;

      .header-link {
        display: flex;
        flex-direction: row;
        align-items: center;
        align-content: space-between;
        font-size: .7rem;
        text-align: right;
        outline: none;
        text-decoration: none;
        width: 9.5rem;
        height: 3rem;

        .header-button-label {
          color: #f5f8fa;
          display: table-cell;
          padding-right: 1rem;
          width: 10rem;
          margin: 0;
        }
      }

      .header-button {
        width: 2.25rem;
        height: 2.25rem;
        margin: auto;
        align-items: center;
      }

      .active .header-button-label {
        font-weight: 600;
        font-size: .75rem;
      }

      &--open {
        transform: translateX(7.65rem);
      }
    }
  }

  &__content {
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
    min-width: 0;
  }

  @media print {
    display: block;
    &__header {
      display: none;
    }
    &__content {
      display: block;
    }
  }

  &__navbar {
    display: flex;
    flex-direction: column;
    height: auto;
    flex: 1;
    padding: 0;

    .navBar__header,
    .navBar__group {
      display: flex;
      padding: .5rem;
    }
    
    .navBar__header {
      text-align: center;
      margin: 0;

      #welcome img {
        width: 2.4rem;
        height: auto;
      };
    }

    .navBar__header + .navBar__group {
      margin-top: 0;
    }

    .navBar__group {
      padding-top: .2rem;
      margin: auto 0;
      flex-direction: column;
      height: auto;
      &:last-of-type {
        margin-bottom: 0;
      }

      ul {
        list-style: none;
        padding: 0;
      }

      li {
        margin: 0 0 1rem;
      }

      button {
        outline: none;
      }
    }

    .header-button {
      img,
      svg {
        width: 16px;
        transition: .3s ease-in filter;
      }
    }

    .log-out .header-button {
      background: radial-gradient(#5a91b3 0, transparent 70%);
      svg {
        filter: invert(.5) sepia(1) saturate(5) hue-rotate(175deg);
      }
    }

    .navBar__group .active .header-button {
      background: radial-gradient(#5a91b3 0, transparent 55%);
      svg {
        filter: invert(.5) sepia(1) saturate(5) hue-rotate(175deg);
      }
    }

    // Below override blueprint
    .bp3-overlay {
      white-space: nowrap;
    }
  }
}

.modal-signin {
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 130;

  &__form {
    max-width: 30rem;
    min-width: 20rem;

    &-buttons {
      display: flex;
      justify-content: center;

      button {
        margin: .25rem;
      }
    }
  }
}

.modal-mentions-legales {
  display: flex;
  justify-content: center;
  align-items: center;
  border: none;
  z-index: 130;

  .bp3-card {
    padding: 0;
  }

  .mentions-legales {
    color: #394b59;
    padding: 1.5rem;

    .mentions-legales-container {
      font-family: "SFProText-Regular", Arial, Helvetica, sans-serif;
    }

    &-title {
      text-align: center;
      font-size: 1.5rem;
      font-weight: 500;
      font-family: "SFProText-Medium", Arial, Helvetica, sans-serif;
      margin: 0 0 2rem 0;
    }

    .container-block {
      > h4 {
        margin-bottom: 0;
        font-family: "SFProText-Medium", Arial, Helvetica, sans-serif;
        padding-bottom: 0.4rem;
      }

      > address {
        font-style: normal;
      }

      p {
        margin-bottom: 0.2rem;
      }

      a {
        color: #394b59;
      }

      a:hover {
        font-weight: bold;
      }
    }
  }
}

@media(min-width: $breakpoints-tablet) and (max-width: $breakpoints-desktop) and (orientation: landscape) {
  .modal-partner {
    >.bp3-card {
      max-width: 50%;
      min-height: 45%;
    }
  }
}

@media(min-width: $breakpoints-phone) and (max-width: $breakpoints-tablet) and (orientation: portrait) {
  .modal-partner {
    .bp3-overlay-content {
      max-width: 90%;
      max-height: 90%;
      overflow: scroll;
    }
  }
}
