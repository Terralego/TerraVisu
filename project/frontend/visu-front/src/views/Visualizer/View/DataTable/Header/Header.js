import React from 'react';
import classnames from 'classnames';
import { Tag, Intent, Position, Popover, PopoverInteractionKind, Button, AnchorButton } from '@blueprintjs/core';
import ColumnsSelector from '@terralego/core/modules/Table/components/ColumnsSelector';

import Loading from '../../../../../components/Loading';

function getIds (features) {
  return features.map(({ _id }) => _id).join(',');
}

const close = (setLayerState, layer) => () => setLayerState({ layer, state: { table: false } });

export const Header = ({
  loading,
  title,
  resultsTotal,
  toggleExtent,
  extent,
  columns,
  onChange,
  resize,
  full,
  exportData,
  selectedFeatures = [],
  compare = '',
  setLayerState,
  displayedLayer: [layer] = [],
}) => (
  <div
    className={classnames({
      'table-header': true,
      'table-header--is-loading': loading,
    })}
  >
    <div className="table-header__title">
      <p className="table-header__title--label">{title}</p>
      <div className="table-header__counter">
        {loading && (
          <Loading
            className="table-header__loading"
            size={20}
          />
        )}
        <Tag
          className="table-header__count"
          intent={Intent.PRIMARY}
          round
        >
          {resultsTotal} résultat{resultsTotal > 1 ? 's' : ''}
        </Tag>
      </div>
    </div>
    <div className="table-header__actions">
      {exportData && (
        <Popover
          content="Exporter"
          interactionKind={PopoverInteractionKind.HOVER}
        >
          <Button
            onClick={exportData}
            icon="export"
            minimal
            intent={Intent.PRIMARY}
            disabled={!resultsTotal || loading}
          />
        </Popover>
      )}
      {compare && (
        <Popover
          content={
            selectedFeatures.length < 2 || selectedFeatures.length > 3
              ? 'Sélectionnez entre 2 et 3 données pour les comparer'
              : 'Comparer ces données'
          }
          interactionKind={PopoverInteractionKind.HOVER}
        >
          <AnchorButton
            target="_blank"
            href={compare.replace(/\{\{ids\}\}/, getIds(selectedFeatures))}
            icon="comparison"
            minimal
            disabled={selectedFeatures.length < 2 || selectedFeatures.length > 3}
            intent={Intent.PRIMARY}
          />
        </Popover>
      )}
      <Popover
        content={extent ? 'Afficher toutes les données' : 'Filtrer selon l\'emprise de la carte'}
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={toggleExtent}
          icon="path-search"
          minimal
          active={extent}
          intent={Intent.PRIMARY}
        />
      </Popover>
      <Popover
        content="Filtrer les propriétés"
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <ColumnsSelector
          columns={columns}
          onChange={onChange}
          position={Position.LEFT}
          locales={{
            displayAllColumns: 'Afficher toutes les colonnes',
            hideAllColumns: 'Cacher toutes les colonnes',
          }}
        />
      </Popover>
      <Popover
        content={full ? 'Réduire' : 'Agrandir'}
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={resize}
          icon={full ? 'minimize' : 'maximize'}
          minimal
          active={full}
          intent={Intent.PRIMARY}
        />
      </Popover>

      <Popover
        content="Fermer"
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={close(setLayerState, layer)}
          icon="cross"
          minimal
          intent={Intent.PRIMARY}
        />
      </Popover>
    </div>
  </div>
);

export default Header;
