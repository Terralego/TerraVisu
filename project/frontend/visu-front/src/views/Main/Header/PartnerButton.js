import React, { useState } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';
import {
  Overlay,
  Classes,
} from '@blueprintjs/core';

import withDeviceSize from '@terralego/core/hoc/withDeviceSize';

import NavBarItemDesktop from '@terralego/core/components/NavBarItemDesktop';
import NavBarItemTablet from '@terralego/core/components/NavBarItemTablet';

import PartnerPage from './PartnerPage';

export const PartnerButton = ({ isMobileSized, isPhoneSized, ...props }) => {
  const [isOpen, setOpen] = useState(false);
  const NavBarItem = isMobileSized ? NavBarItemTablet : NavBarItemDesktop;

  return (
    <>
      <NavBarItem {...props} onClick={() => setOpen(true)} />
      <Overlay
        className={classNames(
          Classes.OVERLAY_SCROLL_CONTAINER,
          Classes.LIGHT,
          'modal-mentions-legales',
        )}
        isOpen={isOpen}
        onClose={() => setOpen(false)}
      >
        <div
          className={classNames(
            Classes.CARD,
            Classes.ELEVATION_4,
          )}
        >
          <PartnerPage />
        </div>
      </Overlay>
    </>
  );
};

PartnerButton.propTypes = {
  isMobileSized: PropTypes.bool,
  isPhoneSized: PropTypes.bool,
};

PartnerButton.defaultProps = {
  isMobileSized: false,
  isPhoneSized: false,
};


export default withDeviceSize()(PartnerButton);
