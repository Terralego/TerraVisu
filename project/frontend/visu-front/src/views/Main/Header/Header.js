import React, { useState, useEffect, useCallback, useMemo } from 'react';

import classnames from 'classnames';
import withDeviceSize from '@terralego/core/hoc/withDeviceSize';
import { connectAuthProvider } from '@terralego/core/modules/Auth';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import MainMenu from '@terralego/core/components/MainMenu';
import LoginButton from '@terralego/core/components/LoginButton';
import { fetchAllViews } from '../../../services/visualizer';

import PartnerButton from './PartnerButton';

import './styles.scss';

const getLinkProps = link => (
  !link.startsWith('http') && {
    link: {
      component: NavLink,
      linkProps: {
        hrefAttribute: 'to',
      },
    },
  }
);

export const Header = ({
  env: { VIEW_ROOT_PATH },
  isHeaderOpen,
  isMobileSized,
  toggleHeader,
  authenticated,
  settings: {
    theme: { logo = '', logoUrl = '/' } = {},
    extraMenuItems = [],
    allowUserRegistration,
  },
}) => {
  const [menu, setMenu] = useState([]);
  const { t } = useTranslation();

  const extraMenuItemsToMenu = useMemo(() => extraMenuItems.map(item => (
    { ...item, ...getLinkProps(item.href) }
  )), [extraMenuItems]);

  const generateMenu = useCallback(views => ({
    navHeader: {
      id: 'welcome',
      label: t('menu.home'),
      href: logoUrl,
      icon: logo,
      ...getLinkProps(logoUrl),
    },
    navItems: [
      views,
      extraMenuItemsToMenu,
      [{
        id: 'nav-partenaires',
        component: () => (
          <PartnerButton
            label={t('menu.informations')}
            icon="info-sign"
          />
        ),
      }, {
        id: 'nav-connexion',
        component: () => (
          <LoginButton
            icon={authenticated ? 'log-out' : 'log-in'}
            label={authenticated ? t('menu.logout') : t('menu.login')}
            className={authenticated ? 'log-out' : 'log-in'}
            translate={t}
            allowUserRegistration={allowUserRegistration}
          />
        ),
      }],
    ],
  }), [t, logoUrl, logo, extraMenuItemsToMenu, authenticated, allowUserRegistration]);


  useEffect(() => {
    let isMounted = true;

    const loadViewList = async () => {
      /** Load the view list and add it to base menu only if component mounted */
      const views = await fetchAllViews(VIEW_ROOT_PATH);
      if (!isMounted) return;
      const viewsToMenu = views.map(view => ({ ...view, ...getLinkProps(view.href) }));
      setMenu(generateMenu(viewsToMenu));
    };

    VIEW_ROOT_PATH && loadViewList();

    return () => {
      isMounted = false;
    };
  }, [VIEW_ROOT_PATH, generateMenu]);

  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <div
      className={classnames(
        'main__header',
        { 'main__header--mobile': isMobileSized },
        { 'main__header--mobile--open': isMobileSized && isHeaderOpen },
      )}
      onClick={() => toggleHeader()}
      role="button"
      tabIndex="-1"
    >
      <MainMenu {...menu} className="main__navbar" />
      {isMobileSized && !isHeaderOpen
        // eslint-disable-next-line jsx-a11y/click-events-have-key-events
        && <div className="main__header__target" role="button" tabIndex="-1" onClick={() => toggleHeader(false)} />
      }
    </div>
  );
};

Header.propTypes = {
  env: PropTypes.shape({
    VIEW_ROOT_PATH: PropTypes.string,
  }),
  isHeaderOpen: PropTypes.bool,
  isMobileSized: PropTypes.bool,
  toggleHeader: PropTypes.func,
  authenticated: PropTypes.bool,
  settings: PropTypes.shape({
    theme: PropTypes.shape({
      logo: PropTypes.string,
      logoUrl: PropTypes.string,
    }),
    extraMenuItems: PropTypes.array,
  }),
};

Header.defaultProps = {
  isHeaderOpen: false,
  isMobileSized: false,
  toggleHeader: () => {},
  authenticated: false,
  settings: {
    theme: {
      logo: '',
      logoUrl: '/',
    },
    extraMenuItems: [],
  },
  env: {
    VIEW_ROOT_PATH: '',
  },
};

export default withDeviceSize()(connectAuthProvider('authenticated')(Header));
