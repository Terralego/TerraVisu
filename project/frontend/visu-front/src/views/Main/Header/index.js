import { connectMain, connectSettings } from '../Provider/context';
import withEnv from '../../../config/withEnv';

import Header from './Header';

export default withEnv(connectMain('isHeaderOpen', 'toggleHeader')(connectSettings('settings')(Header)));
