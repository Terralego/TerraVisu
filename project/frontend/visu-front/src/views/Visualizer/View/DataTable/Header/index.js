import Header from './Header';
import { connectView } from '../../context';

export default connectView(({ setLayerState, layersTreeState }) => ({
  setLayerState,
  displayedLayer: Array
    .from(layersTreeState)
    .find(([, { table }]) => table),
}))(Header);
