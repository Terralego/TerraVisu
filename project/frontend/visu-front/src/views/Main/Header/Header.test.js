import React from 'react';
import { act, create } from 'react-test-renderer';

import { Header } from './Header';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: text => text,
  }),
}));

jest.mock('@terralego/core/components/MainMenu', () => props => <div {...props} />);
jest.mock('@terralego/core/components/LoginButton', () => props => <div {...props} />);

jest.mock('./PartnerButton', () => () => 'PartnerButton');

jest.mock('../../../services/visualizer', () => (
  {
    fetchAllViews: jest.fn(() => [{
      id: 'foo',
      label: 'Foo',
      href: '/foo',
      icon: 'path/to/iconFoo',
    }, {
      id: 'bar',
      label: 'Bar',
      href: '/bar',
      icon: 'path/to/iconBar',
    }]),
  }
));

it('should render correctly', () => {
  let tree;
  act(() => {
    tree = create(
      <Header />,
    );
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render correctly with header open in mobile view', () => {
  let tree;
  act(() => {
    tree = create(
      <Header
        isMobileSized
        isHeaderOpen
      />,
    );
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render correctly with header close in mobile view', () => {
  let tree;
  act(() => {
    tree = create(
      <Header
        location={{ pathname: '/' }}
        env={{ VIEW_ROOT_PATH: '' }}
        settings={{}}
        isMobileSized
        isHeaderOpen={false}
      />,
    );
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

/* it('should open navbar in mobile', () => {
  const toggleHeader = jest.fn();
  const wrapper = shallow(<Header toggleHeader={toggleHeader} />);
  const instance = wrapper.instance();
  const target = '<div class="main__header__target--test" />';

  instance.containerRef.current = {
    target,
    contains: () => true,
  };
  instance.listener({ target });

  expect(instance.props.toggleHeader).toHaveBeenCalledWith();
}); */
