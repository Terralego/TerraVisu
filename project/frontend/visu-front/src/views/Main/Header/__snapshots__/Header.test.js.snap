// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="main__header"
  onClick={[Function]}
  role="button"
  tabIndex="-1"
>
  <div
    className="main__navbar"
  />
</div>
`;

exports[`should render correctly with header close in mobile view 1`] = `
<div
  className="main__header main__header--mobile"
  onClick={[Function]}
  role="button"
  tabIndex="-1"
>
  <div
    className="main__navbar"
  />
  <div
    className="main__header__target"
    onClick={[Function]}
    role="button"
    tabIndex="-1"
  />
</div>
`;

exports[`should render correctly with header open in mobile view 1`] = `
<div
  className="main__header main__header--mobile main__header--mobile--open"
  onClick={[Function]}
  role="button"
  tabIndex="-1"
>
  <div
    className="main__navbar"
  />
</div>
`;
